src/main/java/com/faforever/api/map/MapsController.java

```diff
@@ -61,7 +61,7 @@ public void uploadMap(@RequestParam("file") MultipartFile file,
 
     String extension = Files.getFileExtension(file.getOriginalFilename());
     if (Arrays.stream(fafApiProperties.getMap().getAllowedExtensions()).noneMatch(extension::equals)) {
-      throw new ApiException(new Error(ErrorCode.UPLOAD_INVALID_FILE_EXTENSION, (Object[]) fafApiProperties.getMap().getAllowedExtensions()));
+      throw new ApiException(new Error(ErrorCode.UPLOAD_INVALID_FILE_EXTENSIONS, String.join(", ", fafApiProperties.getMap().getAllowedExtensions())));
```

Comment:
This way, we take the option away from the client to interpret the array of file extension differently. Keep it an array or explain why this is better :)

Reply:
MessageFormat.format(ErrorCode.UPLOAD_INVALID_FILE_EXTENSIONS.getDetail(), (Object[]) new String[]{"png", "jpg"}) == "File must have the following extensions: png."

Reply:
The cast to Object[] should not be necessary.

Reply:
ah now I get it :) I destroyed args.